knowledge_repo/app/index.py

```diff
@@ -124,7 +124,34 @@ def index_up_to_date():
             return False
     return True
 
+@ErrorLog.logged
+def update_index_for_post(kp,path):
+    import time
+    is_index_master = acquire_index_lock()
```

Comment:
@abhirathb Where is this being used?

Reply:
I can possibly check for the flag but since I want this to happen without condition, I don't check for the flag. However, the call to the function anyway gives the control of the lock. So I have received the value of the operation in a variable in case it is useful in the future